### PR TITLE
add NCW feature

### PR DESCRIPTION
**Pull request by amper-fb**
## Pull Request Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes (link to the issue here)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Chore / Documentation
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Locally tested against Fireblocks API

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have added corresponding labels to the PR

### CODE REVIEW

**Review by YoavBZ — COMMENTED** [review]
Thanks @amper-fb for the contribution.
We haven't introduced namespaces in our SDK so far so adding one really changes the UX.
We might better do this in our new SDK and keep the current structure of this SDK.
@dsahni @zoharsf

File: src/api-client.ts [hunk 1]
```diff
@@ -65,6 +65,12 @@ export class ApiClient {
         if (idempotencyKey) {
             headers["Idempotency-Key"] = idempotencyKey;
         }
+
+        const ncwWalletId = requestOptions?.NCW?.walletId;
+        if (ncwWalletId) {
+            headers["X-End-User-Wallet-Id"] = ncwWalletId;
+        }
```
Comment (YoavBZ) [inline src/api-client.ts:72]: ApiClient class is a generic HTTP client wrapper and shouldn't include business logic related functionality

Comment (amper-fb) [reply src/api-client.ts:72]: where do you see business logic here? it's request options mapped to an http header

**Comment by amper-fb** [comment]
> Thanks @amper-fb for the contribution. We haven't introduced namespaces in our SDK so far so adding one really changes the UX. We might better do this in our new SDK and keep the current structure of this SDK. @dsahni @zoharsf

how does it "really change the UX"?

File: src/api-client.ts (hunk 1 above)
Comment (YoavBZ) [reply src/api-client.ts:72]: By business logic I mean that the ApiClient shouldn't be aware of the NCW term.
The `idempotencyKey` is a generic header that affects all endpoints, but I still agree.
Ideally the `headers` should have been given as an optional argument, which could be set in method caller, with the final `Authorization` header addition.

**Comment by YoavBZ** [comment]
> 

We talked about adding namespaces a long time ago and eventually decided not to introduce it in the SDK version (we do that in our next-gen SDK)
Adding a namespace to only one group of methods is confusing and doesn't follow the current structure

**Comment by YoavBZ** [comment]
> > Thanks @amper-fb for the contribution. We haven't introduced namespaces in our SDK so far so adding one really changes the UX. We might better do this in our new SDK and keep the current structure of this SDK. @dsahni @zoharsf
> 
> how does it "really change the UX"?


We talked about adding namespaces a long time ago and eventually decided not to introduce it in the SDK version (we do that in our next-gen SDK)
Adding a namespace to only one group of methods is confusing and doesn't follow the current structure

File: src/api-client.ts (hunk 1 above)
Comment (amper-fb) [reply src/api-client.ts:72]: `X-End-User-Wallet-Id` is a generic header is user together with `Authorization` header generically (recent additions)

**Review by YoavBZ — APPROVED** [review]

**Pull request merged**